cpp/src/arrow/acero/hash_aggregate_test.cc

```diff
@@ -4749,6 +4773,21 @@ TEST_P(GroupBy, PivotDuplicateKeys) {
       RunPivot(key_type, value_type, options, table_json));
 }
 
+TEST_P(GroupBy, PivotInvalidKeys) {
+  // Integer key type, but key names cannot be converted to int
+  auto key_type = int32();
+  auto value_type = float32();
+  std::vector<std::string> table_json = {R"([])"};
+  PivotWiderOptions options(/*key_names=*/{"height", "width"});
+  EXPECT_RAISES_WITH_MESSAGE_THAT(
+      Invalid, HasSubstr("Failed to parse string: 'width' as a scalar of type int32"),
```

Comment:
Picking the last value for the error message is specific to the `Cast` implementation?

Reply:
Hmm, you're right, I should make this test less implementation-dependent.